wire/service.go

```diff
@@ -0,0 +1,91 @@
+package wire
+
+import (
+	"context"
+	"net"
+	"strconv"
+
+	"github.com/evergreen-ci/mrpc"
+	"github.com/evergreen-ci/mrpc/mongowire"
+	"github.com/evergreen-ci/mrpc/shell"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/recovery"
+	"github.com/mongodb/jasper"
+	"github.com/pkg/errors"
+)
+
+// TODO: support jasper.RemoteClient functionality
+type service struct {
```

Comment:
are we worried about thread safety or would a synchronized manager be enough?

Reply:
The thread safety guarantees for this is the same as for the REST/RPC managers. I'm not particularly concerned about having multiple connections given that the synchronized manager already locks all the manager methods.